gonzo/clouds/compute.py

```diff
@@ -126,26 +126,41 @@ def create_instance(self, image_name, name, owner, user_data=None,
         if security_groups is None:
             security_groups = []
 
-        for security_group in security_groups:
-            self.create_if_not_exist_security_group(security_group)
+        for index, security_group in enumerate(security_groups):
+            self.create_if_not_exist_security_group(security_group, subnet_id)
+            security_groups[index] = self.get_security_group(
+                security_group, subnet_id
+            )
 
         security_groups_for_launch = self.security_groups_for_launch(
             security_groups)
 
-        # Availability Zone
-        az = self.get_next_az(environment, server_type)
+        # VPC Specific
+        if subnet_id is not None:
+            az = self.get_next_az(environment, server_type)
+            subnet = self.get_subnet(subnet_id)
+        else:
+            az = None
+            subnet = None
 
         # Launch Instance
-        instance = self.compute_session.create_node(
+        instance_dict = self._generate_instance_dict(
             name=name,
             image=image,
             size=size,
             location=az,
             ex_security_groups=security_groups_for_launch,
+            ex_security_group_ids=security_groups_for_launch,
```

Comment:
do you need both these lines? is `security_groups_for_launch` a list of ids? could suffix the variable name with `_ids` if so to make it super clear.